EventKit/eventkit-cloud
# cleans up unused queues and exchanges

This PR will clean up unused queues and exchanges that remain on the rabbitmq server after new instances of celery are deployed.  

To test:
1. Edit the docker-compose.yml file for the rabbitmq service from
```
expose:
    - "5672"
    - "15672"
```
to
```
ports:
    - "5672:5672"
    - "15672:15672"
```
2. Start the environment and scale celery celery up, and then scale celery down
```
docker-compose up -d --scale celery=3
docker-compose up -d --scale celery=1
```
3.  In a web browser, view the rabbit api and check that there are queues for hosts that are not up anymore `http://guest:guest@cloud.eventkit.test:15672/api/queues`, do the same for exchanges... keep this tab open
4.  Edit the timing for the schedule task `clean-up-queues` in the admin console to happen every minute.
5. Tail the logs and watch for the task to complete.
6. IN a web browser, refresh the view with the api/queues, and verify that the extra unused queues and exchanges were removed.
